types/sqlstring/sqlstring-tests.ts

```diff
@@ -32,3 +33,172 @@ const sql7 = SqlString.format("SELECT * FROM ?? WHERE ?? = ?", inserts);
+assert.equal(SqlString.escape('Super'), "'Super'");
+assert.equal(SqlString.escape('Sup\0er'), "'Sup\\0er'");
+assert.equal(SqlString.escape('Super\0'), "'Super\\0'");
+assert.equal(SqlString.escape('Sup\ber'), "'Sup\\ber'");
+assert.equal(SqlString.escape('Super\b'), "'Super\\b'");
+assert.equal(SqlString.escape('Sup\ner'), "'Sup\\ner'");
+assert.equal(SqlString.escape('Super\n'), "'Super\\n'");
+assert.equal(SqlString.escape('Sup\rer'), "'Sup\\rer'");
+assert.equal(SqlString.escape('Super\r'), "'Super\\r'");
+assert.equal(SqlString.escape('Sup\ter'), "'Sup\\ter'");
+assert.equal(SqlString.escape('Super\t'), "'Super\\t'");
+assert.equal(SqlString.escape('Sup\\er'), "'Sup\\\\er'");
+assert.equal(SqlString.escape('Super\\'), "'Super\\\\'");
+assert.equal(SqlString.escape('Sup\u001aer'), "'Sup\\Zer'");
+assert.equal(SqlString.escape('Super\u001a'), "'Super\\Z'");
+assert.equal(SqlString.escape('Sup\'er'), "'Sup\\'er'");
+assert.equal(SqlString.escape('Super\''), "'Super\\''");
+assert.equal(SqlString.escape('Sup"er'), "'Sup\\\"er'");
+assert.equal(SqlString.escape('Super"'), "'Super\\\"'");
+{
+  const expected = '2012-05-07 11:42:03.002';
+  const date = new Date(2012, 4, 7, 11, 42, 3, 2);
+  const string = SqlString.escape(date);
+  assert.strictEqual(string, "'" + expected + "'");
+}
+{
+  const expected = '2012-05-07 11:42:03.002';
+  const date = new Date(Date.UTC(2012, 4, 7, 11, 42, 3, 2));
+  const string = SqlString.escape(date, false, 'Z');
+  assert.strictEqual(string, "'" + expected + "'");
+}
+{
+  const expected = '2012-05-07 12:42:03.002';
+  const date = new Date(Date.UTC(2012, 4, 7, 11, 42, 3, 2));
+  const string = SqlString.escape(date, false, '+01');
+  assert.strictEqual(string, "'" + expected + "'");
+}
+{
+  const expected = '2012-05-07 13:42:03.002';
+  const date = new Date(Date.UTC(2012, 4, 7, 11, 42, 3, 2));
+  const string = SqlString.escape(date, false, '+0200');
+  assert.strictEqual(string, "'" + expected + "'");
+}
+{
+  const expected = '2012-05-07 06:42:03.002';
+  const date = new Date(Date.UTC(2012, 4, 7, 11, 42, 3, 2));
+  const string = SqlString.escape(date, false, '-05:00');
+  assert.strictEqual(string, "'" + expected + "'");
+}
+{
+  const date = new Date(Date.UTC(2012, 4, 7, 11, 42, 3, 2));
+  const expected = SqlString.escape(date, false, 'Z');
+  const string = SqlString.escape(date, false, 'foo');
+  assert.strictEqual(string, expected);
+}
+{
+  const date = new Date(NaN);
+  const string = SqlString.escape(date);
+  assert.strictEqual(string, 'NULL');
+}
+// {
+//   const buffer = new Buffer([0, 1, 254, 255]);
+//   const string = SqlString.escape(buffer);
+//   assert.strictEqual(string, "X'0001feff'");
+// }
+// {
+//   const buffer = new Buffer([0, 1, 254, 255]);
+//   buffer.toString = function() { return "00' OR '1'='1"; };
+//   const string = SqlString.escape(buffer);
+//   assert.strictEqual(string, "X'00\\' OR \\'1\\'=\\'1'");
+// }
```

Comment:
Buffer requires node type reference. But it does not seem to good from the view of the performance.

Reply:
While I think you could reasonably add a type reference to the test file, you appear to have pretty good test coverage already, so I wouldn't worry about it.